### PR TITLE
227 harun react todo app

### PR DESCRIPTION
**Pull request by ewuharun**
Fixes #227 

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [ ] New feature
- [ ] Refactor
- [x] Infrastructure
- [ ] Other (please specify)

## What was done
created react todo application
## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+
+import { useState } from 'react';
+import Item from './components/Item';
+import TextInputField from './components/Input/TextInputField';
+import './App.css'
+
+function App() {
+
+  const [newTodo, setNewTodo] = useState('');
```
[the remark this answers is not in the document]
Comment (ewuharun) [reply src/App.js:9]: Moved the state into TextInputField component.

File: src/components/Item/Item.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React, { useState } from 'react'
+
+const Item = (props) => {
+    const { name, isActive, id } = props.todo;
+    const [isMouseEnter, setMouseEnter] = useState(false);
```
[the remark this answers is not in the document]
Comment (ewuharun) [reply src/components/Item/Item.js:5]: Changed the naming convention

File: src/components/Item/Item.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    const handleOnMouseLeave = () => {
+        setMouseEnter(false);
+    }
+
+    return (
+        <div className="item-list" onMouseEnter={handleOnMouseEnter} onMouseLeave={handleOnMouseLeave}>
+            <span className={isActive ? 'text-success' : 'line-through'} >{name}</span>
+            <div className={isMouseEnter ? 'btn-group-visible' : 'btn-group-invisible'}>
+                <button
+                    onClick={() => props.doneTodo(id)}
```
[the remark this answers is not in the document]
Comment (ewuharun) [reply src/components/Item/Item.js:20]: changed inline to callback

File: src/components/Item/Item.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                    className="btn-complete">complete
+                </button>
+                <button
+                    onClick={() => props.removeTodo(id)}
```
[the remark this answers is not in the document]
Comment (ewuharun) [reply src/components/Item/Item.js:24]: made it above as callback

**Review by tusher-A — APPROVED** [review]
LGTM

**Pull request merged**